src/storage/v2/vertex_info_cache.cpp

```diff
@@ -0,0 +1,211 @@
+// Copyright 2023 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "storage/v2/vertex_info_cache.hpp"
+
+#include "storage/v2/property_value.hpp"
+
+namespace memgraph::storage {
+
+VertexInfoCache::VertexInfoCache() = default;
+VertexInfoCache::~VertexInfoCache() = default;
+VertexInfoCache::VertexInfoCache(VertexInfoCache &&) noexcept = default;
+VertexInfoCache &VertexInfoCache::operator=(VertexInfoCache &&) noexcept = default;
+
+auto VertexInfoCache::getExists(View view, Vertex const *vertex) const -> std::optional<bool> {
+  auto const &cache = (view == View::OLD) ? old_.existsCache_ : new_.existsCache_;
+  if (cache.empty()) return std::nullopt;
```

Comment:
I expect a find() on an empty map to have similar performance to empty().
Can we just do the find check?

Reply:
`find` will
```cpp
template <class K = key_type>
const_iterator find(const key_arg<K>& key) const {
  prefetch_heap_block();
  return find(key, hash_ref()(key));
}
```
The `empty()` check is a cheap way to avoid what could be a relatively expensive hash.

Reply:
The expectation is the cache rarely has entries, I'm ok with having optimized the expected case.